### PR TITLE
Updated rbd volume example to use yaml configs with better readability

### PR DESCRIPTION
**Pull request by farcaller**
This PR turns barely-readable poorly formatted json to more compact yaml in the examples, that are supposed to be consumed by humans.

(moved from https://github.com/kubernetes/kubernetes/pull/48032)

### CODE REVIEW

File: staging/volumes/rbd/rbd-with-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: rbd2
+spec:
+  containers:
+    - image: kubernetes/pause
+      name: rbd-rw
+      volumeMounts:
+      - {name: rbdpd, mountPath: /mnt/rbd}
```
Comment (ahmetb) [inline staging/volumes/rbd/rbd-with-secret.yaml]: do you mind nesting these instead of using this syntax. it reduces readability.

Comment (farcaller) [inline staging/volumes/rbd/rbd-with-secret.yaml]: Done.

**Comment by sebgoa** [comment]
@farcaller could you squash your commits ?

**Comment by farcaller** [comment]
Done.

**Comment by sebgoa** [comment]
great, thanks for the PR, merging

**Pull request merged**